src/application/services/useHeader.ts

```diff
@@ -0,0 +1,107 @@
+import type { ComputedRef } from 'vue';
+import { computed, onMounted, ref, toRef, watch } from 'vue';
+import { useRoute } from 'vue-router';
+import { AppStateController } from '@/domain';
+import type { Page } from '@/domain/entities/Page';
+import type { TabList } from '@/domain/entities/Tab';
+import { workspaceService } from '@/domain/index';
+import useNote from './useNote';
+import { notEmpty } from '@/infrastructure/utils/empty';
+
+interface useHeaderComposableState {
+  getOpenedPages: () => void;
+
+  addOpenedPage: (page: Page) => void;
+
+  deleteOpenedPage: (page: Page) => void;
+
+  patchPage: (page: Page) => void;
+
+  tabs: ComputedRef<TabList>;
+};
+
+export default function (): useHeaderComposableState {
+  const route = useRoute();
+
+  let noteTitle = ref<string | undefined>(undefined);
+
+  if (route.meta.pageTitle === 'New note') {
+    if (typeof route.params.id === 'string') {
+      const noteId = toRef(route.params.id);
+
+      noteTitle = useNote({ id: noteId }).noteTitle;
+    }
+  }
```

Comment:
looks like a crutch 🤓

Reply:
i guess all route params are defined the same, like string | string[] idk, why, but it seems necessary to check if route.params.id is not string array
![image](https://github.com/codex-team/notes.web/assets/130844513/e1c34e4c-ffbb-416c-86cf-99aa0fd08bc6)

Reply:
moved patching note page title directly to the useNote service